### PR TITLE
[Train] Fix lightning 2.0 import path

### LINKED ISSUE #40260: Release test lightning_gpu_train_3x16_3x1.aws failed
Release test **lightning_gpu_train_3x16_3x1.aws** failed. See https://buildkite.com/ray-project/release-tests-branch/builds/2247#018b1db4-7321-4d03-a82d-51aa0bcea559 for more details.

Managed by OSS Test Policy

### PR DESCRIPTION
**Pull request by woshiyyya**
## Why are these changes needed?

Previously we decide the import path for some components by the lightning/pytorch_lightning version. While if the users installed `pytorch_lightning=2.x.x`, they will not be able to use the new import path `lightning.pytorch`, which caused inconsistency issue.

Now we always import from `lightning.pytorch` first, then fallback to `pytorch_lightning`.

## Related issue number

Closes #40260 #40261 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/train/lightning/_lightning_utils.py [hunk 1]
```diff
@@ -31,10 +31,10 @@ def import_lightning():  # noqa: F402
...
 _TORCH_FSDP_AVAILABLE = _TORCH_GREATER_EQUAL_1_12 and torch.distributed.is_available()
 
-if _LIGHTNING_GREATER_EQUAL_2_0:
+try:
     from lightning.pytorch.plugins.environments import LightningEnvironment
     from lightning.pytorch.strategies import FSDPStrategy
-else:
+except ModuleNotFoundError:
     from pytorch_lightning.plugins.environments import LightningEnvironment
     from pytorch_lightning.strategies import DDPFullyShardedStrategy as FSDPStrategy
```
Comment (matthewdeng) [inline python/ray/train/lightning/_lightning_utils.py:39]: I remember there was a reason we didn't do this, but would it be possible to figure out how to do something like:
```python
    from pl.plugins.environments import LightningEnvironment
    from pl.strategies import FSDPStrategy
```

Comment (woshiyyya) [reply python/ray/train/lightning/_lightning_utils.py:39]: The reason here is we cannot do 
```python
import lightning.pytorch as pl
pl.plugins.environments.LightningEnvironment

# which is similar as
import ray
ray.train.torch.TorchTrainer
```

That's why we explicitly import `LightningEnvironment` here.


```
from pl.plugins.environments import LightningEnvironment
from pl.strategies import FSDPStrategy
```
I remember I searched for quite a while but the python syntax doesn't support using an alias as the prefix of import path.

Comment (woshiyyya) [reply python/ray/train/lightning/_lightning_utils.py:39]: I remember I searched for quite a while, but finally found that the python syntax doesn't support using an alias as the prefix of import path.

```
from pl.plugins.environments import LightningEnvironment
from pl.strategies import FSDPStrategy
```

**Comment by woshiyyya** [comment]
Tests passed: https://buildkite.com/ray-project/release-tests-pr/builds/55457#_

But unfortunately we already deleted these release tests in https://github.com/ray-project/ray/pull/40163

File: python/ray/train/lightning/_lightning_utils.py (hunk 1 above)
Comment (matthewdeng) [reply python/ray/train/lightning/_lightning_utils.py]: Oh that seems a little weird, let me take a look at this today

Comment (matthewdeng) [reply python/ray/train/lightning/_lightning_utils.py]: I think we can just do this? 

```python
LightningEnvironment = pl.plugins.environments.LightningEnvironment

if _LIGHTNING_GREATER_EQUAL_2_0:
    FSDPStrategy = pl.strategies.FSDPStrategy
else:
    FSDPStrategy = pl.strategies.DDPFullyShardedStrategy
```

Comment (woshiyyya) [reply python/ray/train/lightning/_lightning_utils.py]: The FSDP import should be good, but calling `pl.plugins.environments.LightningEnvironment` will raise an error. We can only use `import ... from `clause to use `LightningEnvironment`.

Comment (woshiyyya) [reply python/ray/train/lightning/_lightning_utils.py]: What about we do

```python

try:
    from lightning.pytorch.plugins.environments import LightningEnvironment
except:
    from pytorch_lightning.plugins.environments import LightningEnvironment

if _LIGHTNING_GREATER_EQUAL_2_0:
    FSDPStrategy = pl.strategies.FSDPStrategy
else:
    FSDPStrategy = pl.strategies.DDPFullyShardedStrategy
```

Comment (matthewdeng) [reply python/ray/train/lightning/_lightning_utils.py]: Ah, yeah I guess that seems fine/clean enough, thanks for lookign into this.

**Review by matthewdeng — APPROVED** [review]

**Pull request merged**